feat(workspace): use new doc theme

redallen opened the pull request:
Port of [`theme-patternfly-org`](https://www.npmjs.com/package/theme-patternfly-org) to react. Replaces Gatsby with custom `unified` and `webpack` based build system. New doc features like tabs and [example captions](https://patternfly-pr-3486.surge.sh/components/about-modal/html-demos#basic) are now supported.

Changes:
- **Requires node 12**
- Move bulk selection design guidelines to patternfly.org
- Add new start, serve, and screenshot scripts
  - Update clean task
- Add new full screen example thumbnails
- Fix UMD page
- Remove gatsby and its dependencies
- Upgrade CI to node 12
- Build in CI about ~2m40s faster ([new@1m39s]https://app.circleci.com/pipelines/github/patternfly/patternfly-react/9548/workflows/3cb5cd31-124d-40d1-84a0-e1afb780a703/jobs/43517) / [old@4m19s](https://app.circleci.com/pipelines/github/patternfly/patternfly-react/9508/workflows/c2de2670-4440-47f1-bdf1-87cfd60d6235/jobs/43294))
- Temporarily remove `test:a11y` since theme does not list fullscreen pages in a JSON file anymore. New `test:a11y` script is coming this month from @jschuler and @jessiehuff with improved coverage and reporting.

Related: https://github.com/patternfly/patternfly/pull/3486 and new patternfly.org design!

patternfly-build commented:
PF4 preview: https://patternfly-react-pr-4840.surge.sh

jschuler commented on packages/react-table/src/demos/Table.md:
```diff
@@ -27,56 +16,6 @@ import globalDangerColor200 from '@patternfly/react-tokens/dist/js/global_danger
-import bulkSelectNoToolbarImg from './img/bulk-selection-no-toolbar.png';
-
-## Bulk selection
-Use a bulk selection pattern when you want to select or deselect multiple items in a content view (list, table, or card grid). The bulk selector uses a [Split button](/documentation/react/components/dropdown#split-button) component to control selection from the [Toolbar](/documentation/react/components/toolbar). Besides controlling selection, the bulk selector reflects the selection status of the ...
```
shouldn't this documentation be preserved?

redallen replied:
Yes, it's been moved to https://www.patternfly.org/v4/guidelines/bulk-selection per @mcarrano 's request.

nicolethoen reviewed (commented):
I noticed that the notification badge examples (possibly others) use the `.pf-t-dark` class added so they would have a black background in the examples. 
It doesn't look like the dark theme is being applied anymore.

redallen commented:
@nicolethoen It's not related to that class, but I fixed it!

In the old theme the urls and injected example classNames were based on the file name (`NotificationBadge.md`). They're now based on the `id: Notification badge` and the injected example className is `notification-badge` rather than `notificationbadge`. I fixed most of these about a month ago so that the classes would be correct on patternfly.org, but it looks like I missed this one!

patternfly-build commented:
Your changes have been released in: 
 - @patternfly/react-catalog-view-extension@4.8.35
 - @patternfly/react-charts@6.10.0
 - @patternfly/react-core@4.53.0
 - @patternfly/react-docs@5.9.0
 - @patternfly/react-inline-edit-extension@4.5.91
 - demo-app-ts@4.41.1
 - @patternfly/react-table@4.17.0
 - @patternfly/react-topology@4.6.0
 - @patternfly/react-virtualized-extension@4.5.80

Thanks for your contribution! :tada:

ddonahue007 reviewed (commented):
LGTM
